{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "RESOURCE_LEAK", "qualifier": "resource of type `java.io.DataInputStream` acquired to `is` by call to `new()` at line 42 is not released after line 92.\n**Note**: potential exception at line 47", "severity": "HIGH", "visibility": "user", "line": 92, "column": -1, "procedure": "void CoverageReceiverThread.run()", "procedure_id": "org.pitest.mutationtest.CoverageReceiverThread.run():void.5b45e48704e56ec4010387f2bd295a50", "procedure_start_line": 34, "file": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "bug_trace": [{"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 34, "column_number": -1, "description": "start of procedure run()"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 36, "column_number": -1, "description": ""}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 38, "column_number": -1, "description": "Skipping ServerSocket(...): unknown method"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 39, "column_number": -1, "description": ""}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 41, "column_number": -1, "description": "Skipping BufferedInputStream(...): unknown method"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 42, "column_number": -1, "description": "Skipping DataInputStream(...): unknown method"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 44, "column_number": -1, "description": ""}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 45, "column_number": -1, "description": ""}, {"level": 1, "filename": "pitest/src/main/java/org/pitest/coverage/CoverageStatistics.java", "line_number": 26, "column_number": -1, "description": "start of procedure CoverageStatistics()"}, {"level": 1, "filename": "pitest/src/main/java/org/pitest/coverage/CoverageStatistics.java", "line_number": 28, "column_number": -1, "description": ""}, {"level": 1, "filename": "pitest/src/main/java/org/pitest/coverage/CoverageStatistics.java", "line_number": 29, "column_number": -1, "description": "return from a call to CoverageStatistics.<init>()"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 47, "column_number": -1, "description": "exception java.io.IOException"}, {"level": 0, "filename": "pitest/src/main/java/org/pitest/mutationtest/CoverageReceiverThread.java", "line_number": 92, "column_number": -1, "description": ""}], "key": "CoverageReceiverThread.java|run|RESOURCE_LEAK", "node_key": "f645aafdf4c7de4e5a37ff7e8f2268a3", "hash": "0babd45930a8b4359e5b360d9a0457b1", "bug_type_hum": "Resource Leak", "censored_reason": ""}